src/magicproxy/headers.py

```diff
@@ -0,0 +1,56 @@
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def clean_request_headers(headers, custom_clean_headers):
+    """Removes HTTP Headers for a Request
+
+    Args:
+      headers: the HTTP headers of the request
+      custom_clean_headers: a list of additional headers to remove
+
+    Returns:
+      HTTP headers that have been cleaned of unwanted values
+    """
+    headers = dict(headers)
+    headers.pop("Host", None)
+    headers.pop("Connection", None)
+    # Drop the existing authorization header, it'll only cause problems.
+    headers.pop("Authorization", None)
+    headers = _clean_custom_request_headers(headers, custom_clean_headers)
+    return headers
+
+def clean_response_headers(headers):
+    """Removes HTTP Headers for a Response
+
+    Args:
+      headers: the HTTP headers of the response
+
+    Returns:
+      HTTP headers that have been cleaned of unwanted values
+    """
+    headers = dict(headers)
+    headers.pop("Content-Length", None)
+    headers.pop("Content-Encoding", None)
+    headers.pop("Transfer-Encoding", None)
+    headers["X-Thea-Codes-GitHub-Proxy"] = "1"
+    return headers
+
+
+def _clean_custom_request_headers(headers, custom_clean_headers) -> dict:
```

Comment:
Seems like you can just roll this into the other methods and have two sets `DEFAULT_REMOVED_HEADERS` and `custom_removed_headers` and just use the union of those sets to remove stuff.